docs/notebooks/tutorials/adv_newcv.ipynb

```diff
@@ -5,13 +5,695 @@
+    "## Define the class object\n",
+    "In `mlcolvar` CVs class objects inherit from two mother classes:\n",
+    "- `BaseCV` class, which contains some common and default helper functions\n",
+    "- `lightning.LightniningModule` class, which autoamically gives access to the Lightining package utilities \n",
+    "\n",
+    "In the class declaration preamble we set the names of  the `BLOCKS` that will consitute the main body of the CV itself. These are initialized as a constant for the class and are thus immutable. \n",
+    "\n",
+    "The blocks are meant to correspond to classes and functions defined in `mlcolvar.core` . However, the names we give in `BLOCKS` are arbitrary, considered that, in principle, we could have more blocks of the same types in our model and we would then need to distinguish between them."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "import lightning\n",
+    "\n",
+    "from mlcolvar.cvs import BaseCV\n",
+    "\n",
+    "class AutoEncoderCV(BaseCV, lightning.LightningModule):\n",
+    "    BLOCKS = ['norm_in','encoder','decoder'] "
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To keep the code in the library as clear as possible, we should also add short docstring to our CV class briefly explaining how it works!\n",
+    "\n",
+    "Anyways to save some space we will skip this in the following cells"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class AutoEncoderCV(BaseCV, lightning.LightningModule):\n",
+    "    \"\"\"AutoEncoding Collective Variable. It is composed by a first neural network (encoder) which projects \n",
+    "    the input data into a latent space (the CVs). Then a second network (decoder) takes \n",
+    "    the CVs and tries to reconstruct the input data based on them. It is an unsupervised learning approach, \n",
+    "    typically used when no labels are available.\n",
+    "    Furthermore, it can also be used lo learn a representation which can be used not to reconstruct the data but \n",
+    "    to predict, e.g. future configurations. \n",
+    "\n",
+    "    For training it requires a DictDataset with the key 'data' and optionally 'weights'. If a 'target' \n",
+    "    key is present this will be used as reference for the output of the decoder, otherway this will be compared\n",
+    "    with the input 'data'.\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    BLOCKS = ['norm_in','encoder','decoder'] "
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define the `__init__` method\n",
+    "The `__init__` method is the signature of the CV model as it initializes all that is necessary for the CV model to run, including blocks, variables, loss functions..\n",
```

Comment:
If someone wants to implement his own CV, I think we can assume he knows (or will learn somewhere) how classes in Python work, and keep this section more coincise.

I'd just explain what an autoencoder class needs as an example and say that the ``super().__init__(in_features, out_features)`` must be called here.

Reply:
I tried to slim down this section a little bit in the last commit. Try to check if it makes any sense plz